karapace/schema_registry.py

```diff
@@ -466,6 +466,19 @@ def send_schema_message(
             value = None
         self.producer.send_message(key=key, value=value)
 
+    def is_topic_requiring_validation(self, *, topic_name: TopicName) -> bool:
+        return self.database.is_topic_requiring_validation(topic_name=topic_name)
+
+    def update_require_validation_for_topic(
+        self,
+        *,
+        topic_name: TopicName,
+        skip_validation: bool,
+    ) -> None:
+        key = {"topic": topic_name, "keytype": str(MessageType.schema_validation), "magic": 0}
```

Comment:
The formatting of the key needs to be aligned in the `key_format.py`. This record needs the canonical format, `topic` is an unknown key in data currently.
How does the Confluent Schema Registry react to custom message like this?

Reply:
still a valid point, taking a look tomorrow

Reply:
discussed offline, probably solved. Waiting an update on that

Reply:
I just realized that I need to add the `topic` to the key, otherwise the compaction could delete the topic information, this doesn't change the behaviour of confluent schema registry that is simply to skip the messages he cannot understand (a behaviour that we also do while receiving messages that aren't parsable by us)